src/tests/quantuminspire/qiskit/test_backend_qx.py

```diff
@@ -260,6 +260,32 @@ def test_run_returns_correct_result_for_my_project_number(self):
         api.delete_project.assert_called_with(default_project_number)
         self.assertEqual(my_project_number, int(job.job_id()))
 
+    def test_run_deletes_empty_project_when_error_occurs(self):
+        default_project_number = 42
+        my_project_number = 43
+        api = Mock()
+        api.create_project.return_value = {'id': default_project_number}
+        quantum_inspire_job = Mock()
+        quantum_inspire_job.get_project_identifier.return_value = my_project_number
+        api.execute_qasm_async.return_value = quantum_inspire_job
+        api.get_backend_type_by_name.return_value = {'max_number_of_shots': 4096}
+        simulator = QuantumInspireBackend(api, Mock())
+        instructions = [{'name': 'h', 'qubits': [0]},
```

Comment:
Are all these instructions needed? This is internal qiskit format and prone to change. Could you not write this test by letting qiskit generarate similar data? E.g.
```
c=QuantumCircuit(3,3)
c.h(0)
c.h(2)
c.measure_all()
...
```

Reply:
All these instructions are not necessary, I copied them from another test. Actually a measurement is enough, I changed that.
You are right that this is qiskit format but the easiest way to create sometimes specific instructions for the various test cases. For some test cases this is actually not needed, I added a ticket to the SDK backlog to investigate this suggestion.